.travis.yml

```diff
@@ -8,15 +8,37 @@ notifications:
   - secure: XL7P2xovLrObNNqmWFNcQ+/PIlOJLbixFdqMzrNBm55lXENSwUGgYUeWqULsBWeTHbNknu1XwXDh9EHrziZuocb6ARc1LjAXBT7AzCdAHcvgZRYnnOucm1YWBQcYQJlcUsftVKByVfMyiTvKDw0qYoJZTnzEP5ykBASZvgM3gePcemcZnvAl0jrvSKNpsFwxZt/Yg5wn15COAqbjdKy/4ZXI2QBpZCOX5jHtUy5CGNyKQ9NynC9gcLjgKI1mohYDckTLNP1pdHQ14ecK2PkhOwu92ViRnvCfgNj7gZJSB3jb8JFyMtXZeLuekcJRusvzMI/umctpqHNgqonROTUgJcGyZzkFfXCe1U3SSPENChMW+L3AS/cm6wq7UxLqyyPm8b/6/btvhJE4Dkm/JZjK+d9rTMfRuinij7zlI7yGoP8Z7w3jUJfIMk8mJmJ+/xAueQ8r7YAvImDH4MKduvFgPucdu2jYRNOB7b2wx18c3xjqGz51g3yMhway0vSPwnsHppgmCjLj5FvFYhMFTEoDfQ5fZwhXWfzmTyfh1G7FkcZKcPXWB6WyTQzNJxM1H+TRUnkfiXkd2DIc5d/NSEhzPGOj2GQKT5aoR3xkS7KEpQXF7kzyfs4Abg3WIh7JkYMnONqXDFM4vsE+oQCTygIHiazhh+OG5HnRMqAdiOE2jhg=
   - secure: BS3qr9J4QGVZaI22gQUXOVc722/q8XNSgiply+A65QdsqXLPbgVa3NhxGyIyub3axI+MqyXaXLY1kkO8GOk05zZM3zSl//hvtc3yQKn4PIMOQwSzBJZBpjzwFCRPpyD7JNgHtoF2pya0u/WUqIrOMBRZSvl6/hMoZfTbzDwyxMwbSIHU3fhCwCYKgilKQ42t5hZXQaA4ZSqTyPZ93++AV1C3Y7RkOCzx5i6tIThfvJ/Ky3Xd14LUAmC4lZwBIHF/hjT9rP6oAJQagxEEfbyX+MQz7nTnhPfYrrf8BXcdx8sR9/7zlE+EuPLELzRk8Sy+OEHS6T8npLciSkAsdEey6K51vqTJ/fDyoWhlvTdID/qBJww1OcoRZi+OPUQtHo9kE5sYKADs0HoAXLDFmM4jF8Hd4QQ2ByuhDnGIDveYN9iNw0OQJGDZgbX0QBcvzbhKQN7OPCJkQvKqoixV3X8ejtUkGehBwZAGdhYkVmIeRYB/cSjSzSJx+tRJeOrLqjgUqmehWINIHzkk5qSMtVuhZeRIXDRpDmVYZdiEKFZWpAzAjkXMREDLBC32U4UbOMebrtK6Gz/PwlUQHXkAB4mOpaq3pecKCTvfVGDQMct9lyC6KMsb4EDFn6TPxTzQ4bOPZE6utz7iwQTvXrsFr6g37kzvwczBBkcJX9054OJAypc=
+  - secure: NqG5WvbeIj0JbVrJMbepKZQwHEl2pJSZvYIckDTDy528rGf6ICdmp4PuhwyPboCDYO9/VxySXgaXrGjE3Hc/E9teWqiJwgcnB/YLpVBBgjcEfxOUU+KTUGlx4VJ6dIaQf1I6EBxe8Qw3mWP3gINp89JlO1iqwwKhXPkY2YpMIRZgyHLHcNAiIrRGg2If481XfRDZkVt+k7UfVTYOKygvUncCLv8v02oZ9kWeX61CIkLgqAZYhI8j9bYA0tA+6koHSSZ1IgjvgsmW2CMfjrqPVOwXxnyqM8dFet82BeQtGeWIsYBVtZtgRFCdvpqx7+nQw/addyRVVM1SOir78ZW6mJ2beGT3V5eRAvGNCgGgZDJ4mSIWFrg3Z6EPT3G+zeAzg6dee+kByO++Nx+4cMYIOx9ZUMJC9eOlvrGM44tccvV3HkJ4JVF+vD8gXzDKWh6YaiHPe3iOKy5e5akcWs4WZAvYdfWygJ4bddfCjbO3xMGLSpGQdTiiwiySF22DvyaaBgZcYH6zyYb38KygM9oCMVjeMfDNi8m6qKj6lPNmk1jVxPLq3d30n5biyC2epakMK0waky7JE2h4AYq2Y8cpaXeq9w0gQjtt4QCjTTw56xrk+fRJYcJp2KnCr//GlL5NCdvxG8uu2DqXX6Aeu4Pj+6XPiFUeMQbihrPlDJ8O1zk=
+  - DOCKER_USERNAME=markmcdevops
+jobs:
+  include:
+  - stage: Build docker images
+    before_install:
+    - wget -qO- https://toolbelt.heroku.com/install.sh | sh
+    before_script:
+      - echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin 
+    script:
+    - docker build --target production --tag $DOCKER_USERNAME/todo-app:$TRAVIS_COMMIT .
+    - docker push $DOCKER_USERNAME/todo-app:$TRAVIS_COMMIT
+    
+    after_success:
+    - bash docker_hub_push
+    - heroku container:login
+    - docker pull markmcdevops/todo-app:$TRAVIS_COMMIT
+    - docker tag $DOCKER_USERNAME/todo-app:$TRAVIS_COMMIT registry.heroku.com/todo-app-devops-july/web
+    - docker push registry.heroku.com/todo-app-devops-july/web
+    - heroku container:release web --app todo-app-devops-july
```

Comment:
You are doing this whether or not the tests pass, but shouldn't. You don't want images that are broken.

Also, you are running this for any branch, which is not what you want either. You don't want the version of the app hosted on Heroku to correspond to whichever branch happened to get updated most recently. Make it only run for the master branch  and only do it for branch updates to master, not pull requests. (travis has a `deploy` section you could look into, or you need to put an `if` into this stage)